bigtable/internal/testproxy/proxy.go

```diff
@@ -0,0 +1,963 @@
+	m := mutationFromProto(mPbs)
+
+	t := btc.c.Open(rrq.TableName)
+	row := rrq.RowKey
+
+	ctx = btc.addCancelFunction(ctx)
+
+	err := t.Apply(ctx, string(row), m)
+	if err != nil {
+		return nil, err
+	}
+
+	res := &pb.MutateRowResult{
+		Status: &status.Status{
+			Code: int32(codes.OK),
+		},
+	}
+	return res, nil
+}
+
+// BulkMutateRows responds to the BulkMutateRows RPC. This method applies a
+// series of changes or deletions to multiple rows in a single call.
+func (s *goTestProxyServer) BulkMutateRows(ctx context.Context, req *pb.MutateRowsRequest) (*pb.MutateRowsResult, error) {
+	btc, exists := s.clientIDs[req.ClientId]
+	if !exists {
+		return nil, stat.Error(codes.InvalidArgument,
+			fmt.Sprintf("%s: ClientID does not exist", logLabel))
+	}
+
+	rrq := req.GetRequest()
+	if rrq == nil {
+		return nil, stat.Error(codes.InvalidArgument, "request to BulkMutateRows() is missing inner request")
+	}
+
+	mrs := rrq.Entries
+	t := btc.c.Open(rrq.TableName)
+
+	keys := make([]string, len(mrs))
+	muts := make([]*bigtable.Mutation, len(mrs))
+
+	for i, mr := range mrs {
+
+		key := string(mr.RowKey)
+		m := mutationFromProto(mr.Mutations)
+
+		// A little tricky here ... each key corresponds to a single Mutation
+		// object, where the indices of each slice must be sync'ed.
+		keys[i] = key
+		muts[i] = m
+	}
+
+	log.Printf("keys: %v", keys)
+	log.Printf("mutations: %v", muts)
+
+	ctx = btc.addCancelFunction(ctx)
+
+	errs, err := t.ApplyBulk(ctx, keys, muts)
+	if err != nil {
+		return nil, err
+	}
+
+	entries := make([]*btpb.MutateRowsResponse_Entry, 0)
+
+	// Iterate over any errors returned, matching indices with errors. If
+	// errs is nil, this block is skipped.
+	for i, e := range errs {
+		var me *btpb.MutateRowsResponse_Entry
+		if e != nil {
+			me = &btpb.MutateRowsResponse_Entry{
+				Index: int64(i),
+				Status: &status.Status{
+					Code:    int32(codes.Internal),
+					Message: e.Error(),
+				},
+			}
+
+		}
+		entries = append(entries, me)
+	}
+
+	log.Printf("entries: %v", entries)
+
+	res := &pb.MutateRowsResult{
+		Status: &status.Status{
+			Code: int32(codes.OK),
+		},
+		Entry: entries,
+	}
+
+	return res, nil
+}
+
+// CheckAndMutateRow responds to the CheckAndMutateRow RPC. This method applies
+// one mutation if a condition is true and another mutation if it is false.
+func (s *goTestProxyServer) CheckAndMutateRow(ctx context.Context, req *pb.CheckAndMutateRowRequest) (*pb.CheckAndMutateRowResult, error) {
+	btc, exists := s.clientIDs[req.ClientId]
+	if !exists {
+		return nil, stat.Error(codes.InvalidArgument,
+			fmt.Sprintf("%s: ClientID does not exist", logLabel))
+	}
+
+	rrq := req.GetRequest()
+	if rrq == nil {
+		return nil, stat.Error(codes.InvalidArgument, "request to CheckAndMutateRow() is missing inner request")
+	}
+
+	trueMuts := mutationFromProto(rrq.TrueMutations)
+	falseMuts := mutationFromProto(rrq.FalseMutations)
+
+	rfPb := rrq.PredicateFilter
+	var f bigtable.Filter
+
+	if rfPb != nil {
```

Comment:
Ditto here on the if/else. You can initialize `f` to `bigtablePassAllFilter()` and then override it if need be.

Reply:
Done.